airflow/jobs/triggerer_job_runner.py

```diff
@@ -672,6 +672,17 @@ def update_triggers(self, requested_trigger_ids: set[int]):
                 self.failed_triggers.append((new_id, e))
                 continue
 
+            # If new_trigger_orm.task_instance is None, this means the TaskInstance
+            # row was updated by either Trigger.submit_event or Trigger.submit_failure
+            # and can happen when a single trigger Job is being run on multiple TriggerRunners
+            # in a High-Availability setup.
+            if new_trigger_orm.task_instance is None:
+                self.log.warning((
+                    "TaskInstance for Trigger ID %s is None. It was likely updated by another trigger job. "
+                    "Skipping trigger instantiation."
+                ), new_id)
```

Comment:
Does this warrant a warning? There’s nothing the user can do, and this is kind of a normal thing to happen judging from the original issue.

Reply:
I was trying to follow style of the warning message prior to this, where a warning message is printed if the trigger disappeared before it could be started, which is also normal behavior if the Trigger was cleaned up by a different triggerer.
https://github.com/apache/airflow/blob/6c217ae342e6f86bb4cb448bfd73d5f97cdd41f4/airflow/jobs/triggerer_job_runner.py#L664
Happy to change the logging level if that seems more suitable for this case!